inc/zoo/swar/SWAR.h

```diff
@@ -333,6 +338,10 @@ struct BooleanSWAR: SWAR<NBits, T> {
         SWAR<NBits, T>(initializer)
     {}
 
+    template<int NB, typename TT>
+    friend constexpr auto
+    equals(SWAR<NB, TT>, SWAR<NB, TT>) noexcept;
+
```

Comment:
I don't understand the meaning of declaring an `auto` friend.  Why would this be necessary?

Reply:
I believe you said you preferred the `equals` return type to be auto.